### DISCUSSION TITLE
Multi-can tracking

### DISCUSSION THREAD

**Discussion by Vdol22** [roboflow/supervision#1146] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

I have 6 cams connected in a hallway and my task is to track and count people walking in it (there are always many people there), yet I do not understand how I can produce inference on a multiple cameras AND have the same IDs of people from cam1 to cam2,3...6. I use ultralytics for detection and tried their multi-streaming guide, yet if 1 camera catches a frame without objects - it shuts down. Is there any other way to run inference on multiple cameras or am i missing something? Please help.

**Comment by SkalskiP** [comment]
Hi @Vdol22 👋🏻 Let me start by converting this issue into discussion and moving it to Q&A section.

**Comment by SkalskiP** [comment]
Hi @Vdol22 did you tried to use InferencePipeline`? https://inference.roboflow.com/using_inference/inference_pipeline/#quickstart

**Reply by Vdol22** [reply]
Thanks for the reply! i'll gladly try it, yet is there a way to use yolo weights locally, without Roboflow API? I mean the model_id parameter, which requires Roboflow API key. The machine, which will be used for prediction, won't have Internet connection.

**Reply by SkalskiP** [reply]
You only need to pull this model once. Later it is cached on the machine. Would that work for you?

**Reply by Vdol22** [reply]
Yes, it would. However I struggle to understand how to pull this off (uploading the weights I mean). Let's say I have trained a yolov9c on my PC and would like to use rf inference pipeline. I would still need to create a new project and dataset in my profile, then load my weights and only then download them via InferencePipeline.init(model_id)? Is there any simplier way to just upload the weights and download them with InferencePipeline.init?

Also, will I have to setup a server to use InferencePipeline? The avaible guidelines are pretty vague to me and I'll be very grateful for your assistance.
